modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java

```diff
@@ -44,78 +51,205 @@ public class OrchestratorApi {
     @PostMapping(value = "/register/file",
-            consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE,
             produces = MediaType.APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
     @Transactional
     public Response fetch(@RequestParam(name = "oclLocation") String oclLocation) throws Exception {
-        log.info("Registering managed service with Url {}", oclLocation);
-        this.orchestratorService.registerManagedService(oclLocation);
+        log.info("Register new service with Url {}", oclLocation);
+        registerService.registerServiceByUrl(oclLocation);
         String successMsg = String.format(
-                "Managed service registered with URL %s success.", oclLocation);
+                "Register new service with Url %s success.", oclLocation);
+        log.info(successMsg);
         return Response.successResponse(successMsg);
     }
 
+
     /**
-     * Unregister registered managed service.
+     * Update registered service using id and ocl file url.
      *
-     * @param managedServiceName name of managed service
+     * @param id          id of registered service.
+     * @param oclLocation URL of new Ocl.
      * @return response
      */
-    @Tag(name = "Service Vendor",
-            description = "APIs to publish the managed services.")
-    @DeleteMapping("/register/{managedServiceName}")
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs to manage register services.")
+    @Operation(description = "Update registered service using id and ocl file url.")
+    @PutMapping(value = "/register/file/{id}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
     @Transactional
-    public Response unregister(@PathVariable("managedServiceName") String managedServiceName) {
-        log.info("Unregistering managed service with name {}", managedServiceName);
-        this.orchestratorService.unregisterManagedService(managedServiceName);
+    public Response fetchUpdate(@PathVariable(name = "id") String id,
+            @RequestParam(name = "oclLocation") String oclLocation) throws Exception {
+        log.info("Update registered service {} with Url {}", id, oclLocation);
+        registerService.updateRegisteredServiceByUrl(id, oclLocation);
         String successMsg = String.format(
-                "Managed service %s unregistered success.", managedServiceName);
+                "Update registered service %s with Url %s", id, oclLocation);
+        log.info(successMsg);
         return Response.successResponse(successMsg);
     }
 
+    /**
+     * Unregister registered service using id.
+     *
+     * @param id id of registered service.
+     * @return response
+     */
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs unregister registered service.")
+    @Operation(description = "Unregister registered service using id.")
+    @DeleteMapping("/register/service/{id}")
```

Comment:
```suggestion
    @DeleteMapping("/register/{id}")
```
I think /register/{id} is enought.